QT/sounddisplay.cpp

```diff
@@ -40,9 +63,10 @@ SoundDisplay::~SoundDisplay()
     delete playButton;
     delete timeDomain;
     delete frequencyDisplay;
+    delete effectPanel;
     delete mainLayout;
-    delete domainLayout;
```

Comment:
Where does this get done? It needs to be deleted especially cus now we are adding and removing instances of SoundDisplay